src/cli/commands/upgrade.js

```diff
@@ -27,19 +29,43 @@ export async function run(
     peerDependencies,
   } = await config.readRootManifest() || {};
   const allDependencies = Object.assign({}, peerDependencies, optionalDependencies, devDependencies, dependencies);
+  let addArgs = [];
 
-  const addArgs = args.map((dependency) => {
-    const remoteSource = allDependencies[dependency];
+  if (flags.scope) {
+    flags.scope = flags.scope.replace(/\/?$/g, '/');
+    flags.scope = flags.scope.replace(/^@?/g, '@');
 
-    if (remoteSource && PackageRequest.getExoticResolver(remoteSource)) {
-      return remoteSource;
-    }
+    if (/^@\S*\/$/g.test(flags.scope)) {
+      const searchPattern = new RegExp(`^${flags.scope}`);
```

Comment:
It's better to check directly with `String.prototype.startsWith` (otherwise you'll have issues with the `.` character)